sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -157,13 +151,25 @@ export class InstitutionService extends RecordDataModelService<Institution> {
     return await this.save(institution);
   }
 
+  async syncInstitution(userInfo: UserInfo): Promise<void> {
+    const account = await this.bceidService.getAccountDetails(
+      userInfo.idp_user_name,
+    );
+    const institutionEntity = await this.getInstitute(userInfo);
+    institutionEntity.guid = account.institution.guid;
+    institutionEntity.legalOperatingName = account.institution.legalName;
+    await this.save(institutionEntity);
+    const user = await this.userService.getUser(userInfo.userName);
+    user.firstName = account.user.firstname;
+    user.lastName = account.user.surname;
+    await this.userService.save(user);
+  }
+
   async institutionDetail(userInfo: UserInfo): Promise<InstitutionDetailDto> {
     const account = await this.bceidService.getAccountDetails(
       userInfo.idp_user_name,
     );
-    const institutionEntity = await this.repo.findOne({
-      guid: account.institution.guid,
-    });
+    const institutionEntity = await this.getInstitute(userInfo);
 
     const user = await this.userService.getUser(userInfo.userName);
 
```

Comment:
I think that I missed it in the previous PR, but I believe that the conversion to DTO should happen in the controller layer or anywhere outside this service class.

Reply:
I like mapper concept, but that require additional learning, so skip this for current works. Mapper could be part of entity.

Reply:
This code is outdated, I don't know why git is not reflecting that

Reply:
I don't agree for controller, but I agree there should be a mapping layer.

Reply:
But mapping logic is there from the beginning, in this PR there is little change on that portion. Refactoring them will be totally change the task estimation.